### ISSUE TITLE
Support babylon.js v5.0.0-dev

### ISSUE THREAD

**Issue by il-m-yamagishi** [virtual-cast/babylon-mtoon-material#66]
- babylon.js v5.0.0-alpha.25
- WebGL2

Althouth [the meshUboDeclaration shader](https://github.com/BabylonJS/Babylon.js/blob/master/src/Shaders/ShadersInclude/meshUboDeclaration.fx) defines `#define WORLD_UBO`, [instancesDeclaration.fx](https://github.com/BabylonJS/Babylon.js/blob/master/src/Shaders/ShadersInclude/instancesDeclaration.fx) outputs `uniform mat4 world;`...?

```
logger.js:84 BJS - [21:41:17]: Offending line [222] in vertex code: out vec2 vMainUV1;
logger.js:84 BJS - [21:41:17]: Error: VERTEX SHADER ERROR: 0:222: 'world' : redefinition
```

The code near the error above.

[Error Line](https://github.com/virtual-cast/babylon-mtoon-material/blob/ea6e549abe99d6df312ed94101a555c8983913c9/src/shaders/mtoon.vert#L28)

```glsl
// Uniforms
uniform mat4 world; // problem point is here. line 222
out vec2 vMainUV1;
```

- In WebGL1, this shader is correctly rendered.

<details>
  <summary>Generated Vertex Code</summary>
<p>

```glsl
#version 300 es
#define WEBGL2 
#define CUSTOMUSERLIGHTING
#define MTOON_OUTLINE_WIDTH_WORLD
#define MTOON_OUTLINE_COLOR_MIXED
#define SHADE
#define SHADEDIRECTUV 1
#define RECEIVE_SHADOWDIRECTUV 0
#define SHADING_GRADEDIRECTUV 0
#define RIMDIRECTUV 0
#define MATCAPDIRECTUV 0
#define OUTLINE_WIDTHDIRECTUV 0
#define UV_ANIMATION_MASKDIRECTUV 0
#define MAINUV1
#define DIFFUSE
#define DIFFUSEDIRECTUV 1
#define DETAILDIRECTUV 0
#define DETAIL_NORMALBLENDMETHOD 0
#define EMISSIVEDIRECTUV 0
#define BUMPDIRECTUV 0
#define NORMAL
#define UV1
#define NUM_BONE_INFLUENCERS 0
#define BonesPerMesh 0
#define NUM_MORPH_INFLUENCERS 0
#define ALPHATEST_AFTERALLALPHACOMPUTATIONS
#define ALPHABLEND
#define PREPASS_IRRADIANCE_INDEX -1
#define PREPASS_ALBEDO_INDEX -1
#define PREPASS_DEPTH_INDEX -1
#define PREPASS_NORMAL_INDEX -1
#define PREPASS_POSITION_INDEX -1
#define PREPASS_VELOCITY_INDEX -1
#define PREPASS_REFLECTIVITY_INDEX -1
#define SCENE_MRT_COUNT 0
#define LIGHT0
#define DIRLIGHT0

#define SHADER_NAME vertex:mtoon
precision highp float;
// この include は特別で、 UboDeclaration または VertexDeclaration のどちらかに置換される
// @see effect.ts
// include<__decl__mtoonVertex> または include<__decl__mtoonFragment> と書いた時に WebGL2 の場合展開される
// @see effect.ts
layout(std140, column_major) uniform;
uniform Material {
// Color & Texture
    vec4 vDiffuseColor;
    vec2 vDiffuseInfos;
    mat4 diffuseMatrix;
    vec4 vEmissiveColor;
    vec2 vEmissiveInfos;
    mat4 emissiveMatrix;
    vec3 vBumpInfos;
    mat4 bumpMatrix;
    vec3 vShadeColor;
    vec2 vShadeInfos;
    mat4 shadeMatrix;
    vec2 vReceiveShadowInfos;
    mat4 receiveShadowMatrix;
    vec2 vShadingGradeInfos;
    mat4 shadingGradeMatrix;
    vec3 vRimColor;
    vec2 vRimInfos;
    mat4 rimMatrix;
    vec2 vMatCapInfos;
    mat4 matCapMatrix;
    vec4 vOutlineColor;
    vec2 vOutlineWidthInfos;
    mat4 outlineWidthMatrix;
    vec2 vUvAnimationMaskInfos;
    mat4 uvAnimationMaskMatrix;
// babylon specific
    vec2 vTangentSpaceParams;
    float pointSize;
// MToon params
    float shadingGradeRate;
    float receiveShadowRate;
    float shadeShift;
    float shadeToony;
    float lightColorAttenuation;
    float indirectLightIntensity;
    float rimLightingMix;
    float rimFresnelPower;
    float rimLift;
    float outlineWidth;
    float outlineScaledMaxDistance;
    float outlineLightingMix;
    float uvAnimationScrollX;
    float uvAnimationScrollY;
    float uvAnimationRotation;
    vec3 vEyeUp;
    float alphaCutOff;
    vec3 vAmbientColor;
    float aspect;
    float isOutline;
    vec4 time;
};
// babylon specific
layout(std140, column_major) uniform;
uniform Scene {
    mat4 viewProjection;
    mat4 view;
    mat4 projection;
    vec4 vEyePosition;
};
layout(std140, column_major) uniform;
uniform Mesh {
    mat4 world; // ****uniform world is declared here****
    float visibility;
};
#define WORLD_UBO

// 基本的に default.vertex.fx を踏襲している
// Attributes
in vec3 position;
in vec3 normal;
in vec2 uv;
const float PI = 3.1415926535897932384626433832795;
const float HALF_MIN = 5.96046448e-08;
const float LinearEncodePowerApprox = 2.2;
const float GammaEncodePowerApprox = 1.0 / LinearEncodePowerApprox;
const vec3 LuminanceEncodeApprox = vec3(0.2126, 0.7152, 0.0722);
const float Epsilon = 0.0000001;
#define saturate(x) clamp(x,0.0,1.0)
#define absEps(x) abs(x)+Epsilon
#define maxEps(x) max(x,Epsilon)
#define saturateEps(x) clamp(x,Epsilon,1.0)
mat3 transposeMat3(mat3 inMatrix) {
    vec3 i0 = inMatrix[0];
    vec3 i1 = inMatrix[1];
    vec3 i2 = inMatrix[2];
    mat3 outMatrix = mat3(vec3(i0.x, i1.x, i2.x), vec3(i0.y, i1.y, i2.y), vec3(i0.z, i1.z, i2.z));
    return outMatrix;
}
mat3 inverseMat3(mat3 inMatrix) {
    float a00 = inMatrix[0][0], a01 = inMatrix[0][1], a02 = inMatrix[0][2];
    float a10 = inMatrix[1][0], a11 = inMatrix[1][1], a12 = inMatrix[1][2];
    float a20 = inMatrix[2][0], a21 = inMatrix[2][1], a22 = inMatrix[2][2];
    float b01 = a22 * a11 - a12 * a21;
    float b11 = -a22 * a10 + a12 * a20;
    float b21 = a21 * a10 - a11 * a20;
    float det = a00 * b01 + a01 * b11 + a02 * b21;
    return mat3(b01, (-a22 * a01 + a02 * a21), (a12 * a01 - a02 * a11), b11, (a22 * a00 - a02 * a20), (-a12 * a00 + a02 * a10), b21, (-a21 * a00 + a01 * a20), (a11 * a00 - a01 * a10)) / det;
}
float toLinearSpace(float color) {
    return pow(color, LinearEncodePowerApprox);
}
vec3 toLinearSpace(vec3 color) {
    return pow(color, vec3(LinearEncodePowerApprox));
}
vec4 toLinearSpace(vec4 color) {
    return vec4(pow(color.rgb, vec3(LinearEncodePowerApprox)), color.a);
}
vec3 toGammaSpace(vec3 color) {
    return pow(color, vec3(GammaEncodePowerApprox));
}
vec4 toGammaSpace(vec4 color) {
    return vec4(pow(color.rgb, vec3(GammaEncodePowerApprox)), color.a);
}
float toGammaSpace(float color) {
    return pow(color, GammaEncodePowerApprox);
}
float square(float value) {
    return value * value;
}
float pow5(float value) {
    float sq = value * value;
    return sq * sq * value;
}
float getLuminance(vec3 color) {
    return clamp(dot(color, LuminanceEncodeApprox), 0., 1.);
}
float getRand(vec2 seed) {
    return fract(sin(dot(seed.xy, vec2(12.9898, 78.233))) * 43758.5453);
}
float dither(vec2 seed, float varianceAmount) {
    float rand = getRand(seed);
    float dither = mix(-varianceAmount / 255.0, varianceAmount / 255.0, rand);
    return dither;
}
const float rgbdMaxRange = 255.0;
vec4 toRGBD(vec3 color) {
    float maxRGB = maxEps(max(color.r, max(color.g, color.b)));
    float D = max(rgbdMaxRange / maxRGB, 1.);
    D = clamp(floor(D) / 255.0, 0., 1.);
    vec3 rgb = color.rgb * D;
    rgb = toGammaSpace(rgb);
    return vec4(clamp(rgb, 0., 1.), D);
}
vec3 fromRGBD(vec4 rgbd) {
    rgbd.rgb = toLinearSpace(rgbd.rgb);
    return rgbd.rgb / rgbd.a;
}
vec3 parallaxCorrectNormal(vec3 vertexPos, vec3 origVec, vec3 cubeSize, vec3 cubePos) {
    vec3 invOrigVec = vec3(1.0, 1.0, 1.0) / origVec;
    vec3 halfSize = cubeSize * 0.5;
    vec3 intersecAtMaxPlane = (cubePos + halfSize - vertexPos) * invOrigVec;
    vec3 intersecAtMinPlane = (cubePos - halfSize - vertexPos) * invOrigVec;
    vec3 largestIntersec = max(intersecAtMaxPlane, intersecAtMinPlane);
    float distance = min(min(largestIntersec.x, largestIntersec.y), largestIntersec.z);
    vec3 intersectPositionWS = vertexPos + origVec * distance;
    return intersectPositionWS - cubePos;
}
// Uniforms
uniform mat4 world; // ****problem point is here. line 222****
out vec2 vMainUV1;
// Output
out vec3 vPositionW;
out vec3 vNormalW;
uniform Light0 {
    vec4 vLightData;
    vec4 vLightDiffuse;
    vec4 vLightSpecular;
    vec4 shadowsInfo;
    vec2 depthValues;
} light0;
// Additional Uniforms
void main(void) {
    vec3 positionUpdated = position;
    vec3 normalUpdated = normal;
    mat4 finalWorld = world;
// Texture coordinates
    vec2 uv2 = vec2(0., 0.);
    vMainUV1 = uv;
    float outlineTex = 1.0;
    if(isOutline == 1.0) {
// ワールド座標の normal 分だけ移動する
        vec3 outlineOffset = normalize(finalWorld * vec4(normalUpdated, 1.0)).xyz * 0.01 * outlineWidth * outlineTex;
        positionUpdated.xyz += outlineOffset;
    } // End isOutline == 1.0
    vec4 vertex = vec4(1.0);
    vertex = viewProjection * finalWorld * vec4(positionUpdated, 1.0);
    if(isOutline == 1.0) {
        vertex.z += 1E-2 * vertex.w;
// anti-artifact magic from three-vrm
    }
    gl_Position = vertex;
    vec4 worldPos = finalWorld * vec4(positionUpdated, 1.0);
    vPositionW = vec3(worldPos);
    mat3 normalWorld = mat3(finalWorld);
    vNormalW = normalize(normalWorld * normalUpdated);
}
```

</p>
</details>

**Comment by il-m-yamagishi** [comment]
reported to babylon.js forum.

see -> https://forum.babylonjs.com/t/vertex-shader-error-0-world-redefinition-errors-appeared-in-my-own-shader-mtoonmaterial/21665

**Comment by il-m-yamagishi** [comment]
The problem was solved at https://forum.babylonjs.com/t/vertex-shader-error-0-world-redefinition-errors-appeared-in-my-own-shader-mtoonmaterial/21665/6?u=il-m-yamagishi

**Comment by il-m-yamagishi** [comment]
- closed by #83
